packages/core/src/__wallet__/web/connector/wallet-connect/base.ts

```diff
@@ -0,0 +1,475 @@
+import {
+  uint8ArrayFromHexString,
+  uint8ArrayToHexString,
+  stringToBase64urlString,
+} from "@turnkey/encoding";
+import bs58 from "bs58";
+import { recoverPublicKey, hashMessage, type Hex, toHex } from "viem";
+import { compressRawPublicKey } from "@turnkey/crypto";
+import {
+  Chain,
+  WalletInterfaceType,
+  WalletProvider,
+  WalletProviderInfo,
+  SignIntent,
+  WalletConnectProvider,
+  WalletConnectInterface,
+  SwitchableChain,
+} from "@types";
+import type { WalletConnectClient } from "./client";
+import type { SessionTypes } from "@walletconnect/types";
+import { Transaction } from "ethers";
+
+export class WalletConnectWallet implements WalletConnectInterface {
+  readonly interfaceType = WalletInterfaceType.WalletConnect;
+
+  private ethereumNamespaces: string[] = [];
+  private solanaNamespaces: string[] = [];
+
+  private ethChain!: string;
+  private solChain!: string;
+
+  private uri?: string;
+
+  constructor(private client: WalletConnectClient) {
+    this.client.onSessionDelete(() => {});
+  }
+
+  /**
+   * Initializes WalletConnect pairing flow with the specified chains.
+   * If an active session already exists, pairing is skipped.
+   *
+   * @param opts.ethereum - Whether to enable Ethereum pairing
+   * @param opts.solana - Whether to enable Solana pairing
+   * @throws {Error} If no chains are enabled
+   */
+  async init(opts: {
+    ethereumNamespaces: string[];
+    solanaNamespaces: string[];
+  }): Promise<void> {
+    this.ethereumNamespaces = opts.ethereumNamespaces;
+    if (this.ethereumNamespaces.length > 0) {
+      this.ethChain = this.ethereumNamespaces[0]!;
+    }
+
+    this.solanaNamespaces = opts.solanaNamespaces;
+    if (this.solanaNamespaces.length > 0) {
+      this.solChain = this.solanaNamespaces[0]!;
+    }
+
+    if (
+      this.ethereumNamespaces.length === 0 &&
+      this.solanaNamespaces.length === 0
+    ) {
+      throw new Error(
+        "At least one namespace must be enabled for WalletConnect",
+      );
+    }
+
+    const session = this.client.getSession();
+    if (hasConnectedAccounts(session)) {
+      return;
+    }
+
+    const namespaces: Record<string, any> = {};
```

Comment:
nonblocking nit: would be nice to replace the `any` with some predefined type possibilities (`eip155, solana`, etc)